pageserver/src/import_datadir.rs

```diff
@@ -725,10 +663,10 @@ pub async fn create_tar_zst(pgdata_path: &Utf8Path) -> Result<Vec<u8>> {
     let mut zstd = builder.into_inner().await?;
     zstd.shutdown().await?;
     let compressed = zstd.into_inner();
-    let compressed_len = compressed.buf.len();
-    const INITDB_TAR_ZST_WARN_LIMIT: usize = 2_000_000;
+    let compressed_len = compressed.metadata().await?.len();
```

Comment:
Would be nice if we could avoid the `.metadata()` call by making the zstd builder return the final size on `.shutdown().await`

Reply:
can you give concrete code suggestions? `.shutdown().await` has `Result<()>` as the return value. I don't see a way outside of creating a wrapper for the file that tracks the written offsets, which would involve a lot of repetitive code.

Reply:
resolving because I remember dimly that you said that the metadata call was needed.